File: 2023-2024/sourcecode/mod3/lab11/RoomBasicsJava/app/src/main/java/com/example/roombasicsjava/ui/UserComparator.java
package com.example.roombasicsjava.ui;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import com.example.roombasicsjava.data.User;

public class UserComparator extends DiffUtil.ItemCallback<User> {
    @Override
    public boolean areItemsTheSame(@NonNull User oldItem, @NonNull User newItem) {
        return oldItem == newItem;
    }

    @Override
    public boolean areContentsTheSame(User oldItem, @NonNull User newItem) {
        return oldItem.equals(newItem);
    }
}

File: 2023-2024/sourcecode/mod3/lab11/RoomBasicsKotlin/app/src/main/java/com/example/roombasicskotlin/ui/UserComparator.kt
package com.example.roombasicskotlin.ui

import androidx.recyclerview.widget.DiffUtil
import com.example.roombasicskotlin.data.User

class UserComparator : DiffUtil.ItemCallback<User>() {
    override fun areItemsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem === newItem
    }

    override fun areContentsTheSame(oldItem: User, newItem: User): Boolean {
        return oldItem == newItem
    }
}